### ISSUE TITLE
[FEATURE REQ] How to delete KeyVault secret versions?

### ISSUE THREAD

**Issue by xperiandri** [Azure/azure-sdk-for-net#20934]
**Library or service name.**
`Azure.Security.KeyVault.Secrets`
 
**Is your feature request related to a problem? Please describe.**
How can I programmatically delete a secret version that already expired?

**Comment by xperiandri** [comment]
But it is possible to do that on the portal

**Comment by jlichwa** [comment]
@xperiandri there is no option to delete version , you need to delete entire secret.

**Comment by xperiandri** [comment]
Indeed. What is the solution if I have too many old secret versions?

**Comment by heaths** [comment]
There is no way to delete older versions apart from deleting the entire secret and recreating, or creating a new secret name. Can you describe why there are "too many"?

**Comment by jlichwa** [comment]
It is correct you will need to create new secret with new name (copy few versions if needed). @heaths the issue is that i.e. backup works only for up to 500 versions, after that there is no way to backup a secret.

**Comment by xperiandri** [comment]
Do you mean that it works like a stack? After reaching 500 versions each new version will push put the oldest version?

**Comment by jlichwa** [comment]
@xperiandri it's not a stack, you will be able to see and access all versions, it's just after 500 versions the backup operation will time out.

**Comment by xperiandri** [comment]
So if I don't need a backup don't I need to worry about that?

**Comment by jlichwa** [comment]
You may notice degradation of performance over time and we don't recommend it. All operations besides backup will continue work though. Is there particular scenario that you need all versions?

**Comment by xperiandri** [comment]
I store refresh token to another system per user as Key Vault secret

**Comment by xperiandri** [comment]
So I need to create a Key Vault without a restore (like recycler bin) feature. Then I can delete a whole secret if I have more than 20 versions and create a new one in place?

**Comment by jlichwa** [comment]
The soft-delete (recycler bin) soon will enforced, so that will not work longer term, you will need to delete and purge and create new one.  
Deleting old versions is valid feature request, we will move that request to service side (rest api).
